ouroboros-network/src/Ouroboros/Network/MockChain/Chain.hs

```diff
@@ -119,8 +121,8 @@ validExtension
   => Chain block -> block -> Bool
 validExtension c b = blockInvariant b
                   && headHash c == blockPrevHash b
-                  -- The Ord instance for WithOrigin puts At _ after Origin.
                   && headSlot c <  At (blockSlot b)
+                  -- TODO This is a hack which should be addressed through 'WithOrigin'
```

Comment:
I'm not sure we'll actually address this with `WithOrigin`, see https://github.com/input-output-hk/ouroboros-network/pull/774#issuecomment-513234912